src/access_list.rs

```diff
@@ -65,67 +71,70 @@ impl AccessListInspector {
     /// top-level call.
     ///
     /// Those include caller, callee and precompiles.
-    fn collect_excluded_addresses<DB: Database>(&mut self, context: &EvmContext<DB>) {
-        let from = context.env.tx.caller;
-        let to = if let TxKind::Call(to) = context.env.tx.transact_to {
+    fn collect_excluded_addresses<CTX: ContextTrait<Journal: JournalExt>>(
+        &mut self,
+        context: &CTX,
+    ) {
+        let from = context.tx().caller();
+        let to = if let TxKind::Call(to) = context.tx().kind() {
             to
         } else {
             // We need to exclude the created address if this is a CREATE frame.
             //
             // This assumes that caller has already been loaded but nonce was not increased yet.
-            let nonce = context.journaled_state.account(from).info.nonce;
+            let nonce = context.journal_ref().evm_state().get(&from).unwrap().info.nonce;
             from.create(nonce)
         };
-        let precompiles = context.precompiles.addresses().copied();
+        let precompiles = context.journal_ref().precompile_addresses().clone();
         self.excluded = [from, to].into_iter().chain(precompiles).collect();
     }
 }
 
-impl<DB> Inspector<DB> for AccessListInspector
+impl<CTX> Inspector<CTX, EthInterpreter> for AccessListInspector
 where
-    DB: Database,
+    CTX: ContextTrait<Journal: JournalExt>,
```

Comment:
naming a trait Trait always feels a bit weird to me

Reply:
Will rename all of them to `ContextT`.